src/flamegraph/attrs.rs

```diff
@@ -205,51 +202,36 @@ mod test {
         let r = s.as_bytes();
 
         let mut expected_inner = HashMap::new();
-        let foo_g_extra: Vec<(String, String)> = vec![
-            ("gextra1".to_owned(), "gextra1".to_owned()),
-            ("gextra2".to_owned(), "foo gextra2".to_owned()),
-        ];
-        let foo_a_extra: Vec<(String, String)> = vec![
-            ("aextra1".to_owned(), "foo aextra1".to_owned()),
-            ("aextra2".to_owned(), "foo aextra2".to_owned()),
-        ];
+        let foo_attrs: IndexMap<String, String> = indexmap! {
+            "class".to_owned() => "foo class".to_owned(),
+            "xlink:href".to_owned() => "foo href".to_owned(),
+            "target".to_owned() => "foo target".to_owned(),
+            "gextra1".to_owned() => "gextra1".to_owned(),
+            "gextra2".to_owned() => "foo gextra2".to_owned(),
+            "aextra1".to_owned() => "foo aextra1".to_owned(),
+            "aextra2".to_owned() => "foo aextra2".to_owned(),
+        };
 
         expected_inner.insert(
             "foo".to_owned(),
             FrameAttrs {
                 title: Some("foo title".to_owned()),
-                g: GElementAttrs {
-                    id: None,
-                    class: Some("foo class".to_owned()),
-                    extra: foo_g_extra,
-                },
-                a: AElementAttrs {
-                    href: Some("foo href".to_owned()),
-                    target: Some("foo target".to_owned()),
-                    extra: foo_a_extra,
-                },
+                attrs: foo_attrs,
             },
         );
 
-        let bar_a_extra: Vec<(String, String)> = vec![
-            ("aextra1".to_owned(), "foo".to_owned()),
-            ("aextra2".to_owned(), "bar".to_owned()),
+        let bar_attrs: IndexMap<String, String> = indexmap![
```

Comment:
If you just want the macro, you can just use the [`maplit` crate](https://docs.rs/maplit/).